components/browser/search/src/main/java/mozilla/components/browser/search/suggestions/SearchSuggestionClient.kt

```diff
@@ -0,0 +1,38 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.search.suggestions
+
+import mozilla.components.browser.search.SearchEngine
+
+/**
+ * Async function responsible for taking a URL and returning the results
+ */
+typealias SearchSuggestionFetcher = suspend (url: String) -> String?
```

Comment:
Do you want to leave it to the app to implement this? Do you expect apps have very different implementations of it? Should we provide a simple default implementation based on HttpUrlConnection? I assume you will need to write something for Focus anyways?

Reply:
that was the idea. I'm not opposed to including a default fetcher (I'm also not a huge fan of the name the more I say it.. so suggestions are definitely welcome!)